From 5b6c595d71f2091039d71b858c68f8f05e982bd8 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Sat, 29 Jul 2017 11:34:45 +0200
Subject: [PATCH] Disable the GDK thread locking

Without it the rendering does not work and the following error is
periodically shown on stdout:
err:ntdll:RtlpWaitForCriticalSection section 0x24f230 "?" wait timed out in thread 0009, blocked by 0000, retrying (60 sec)

See further details in the relevant issue:
https://github.com/pavouk/lgi/issues/98
---
 lgi/override/Gdk.lua | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lgi/override/Gdk.lua b/lgi/override/Gdk.lua
index 7a68ffe..d76b3ed 100644
--- a/lgi/override/Gdk.lua
+++ b/lgi/override/Gdk.lua
@@ -19,10 +19,6 @@ local ti = ffi.types
 local Gdk = lgi.Gdk
 local cairo = lgi.cairo
 
--- Take over internal GDK synchronization lock.
-core.registerlock(core.gi.Gdk.resolve.gdk_threads_set_lock_functions)
-Gdk.threads_init()
-
 -- Gdk.Rectangle does not exist at all in older GOI, because it is
 -- aliased to cairo.RectangleInt.  Make sure that we have it exists,
 -- because it is very commonly used in API documentation.
-- 
2.13.3

